### DISCUSSION TITLE
Trying to change semi-detached binary hierarchy to contact like.

### DISCUSSION THREAD

**Discussion by mousam100** [phoebe-project/phoebe2#835] [General]
Hello developers,
My name is Mousam, and I am very new to Phoebe. I am encountering a problem and would appreciate your suggestions on it. I am attempting to develop a system with parameters such as mass ratio (q), period, temperature of the primary and secondary stars, and semi-major axis (sma) in a semi-detached binary configuration. However, I am finding it challenging to freely adjust these parameters due to constraints inherent in semi-detached binaries. I want to achieve a configuration with a very short period and a small semi-major axis while maintaining the semi-detached nature of the system. Can I achieve this by changing the semi-detached hierarchy to a contact configuration? 

When I attempt to change the hierarchy of the semi-detached binary to contact, I encounter errors. Please guide me regarding this. I have provided the code for your reference along with the encountered errors. Thank you for your assistance.
 
[Semi_Detached001.pdf](https://github.com/phoebe-project/phoebe2/files/14644083/Semi_Detached001.pdf)

**Comment by kecnry** [comment]
It technically is possible (although difficult) to change a bundle from semi-detached to contact, but since you're writing a script anyways, it will be much easier to just modify the call to [default_binary](http://phoebe-project.org/docs/latest/api/phoebe.frontend.bundle.Bundle.default_binary).  For example the following:

```
phoebe.default_binary()
phoebe.default_binary(contact=True)
phoebe.default_binary(semidetached='primary')
phoebe.default_binary(semidetached='secondary')
```

will setup a detached, contact, semidetached where the primary component is at the critical radius, semidetached where the secondary component is at the critical radius, respectively.

If you _know_ that one of your stars is filling its roche lobe, then using the semi-detached constraint makes sense, but generally I'd suggest either using a detached or contact system from the start.  A detached system will let a star reach the critical radius (but prevent it from overflowing), whereas a contact system will not allow dropping below the critical value.  A "semi-detached" system will enforce one (or both) of the stars to be exactly at the critical radius (completely filling the roche lobe).

Does this help clarify anything at all?

-Kyle

**Reply by mousam100** [reply]
Thank you Kyle.
